src/main/java/EventsAndCommands/CommandLoader.java

```diff
@@ -135,30 +102,42 @@ public Command injectCommand(Constructor<?> ctor) throws Exception{
             }
         }
 
-        return (Command) ctor.newInstance(parameters);
+        return (T) ctor.newInstance(parameters);
     }
 
-    public boolean hasAcceptableParams(Constructor<?> ctor){
+    /**
+     * Finds the ctor with the most args that fits the Dependency Injection
+     * criteria. The constructor needs to be empty or have any combination of the
+     * following parametertypes: {{@see MongoDatabase}, {@see Properties},
+     * {@see EventWaiter}}.
+     * 
+     * @param cl The class that has constructors.
+     * @return The best fitting constructor.
+     */
+    private Constructor<? extends T> findFittingConstructor(Class<? extends T> cl) {
         final Set<Class<?>> acceptableParams = new HashSet<>();
         acceptableParams.add(MongoDatabase.class);
         acceptableParams.add(Properties.class);
         acceptableParams.add(EventWaiter.class);
 
-        return acceptableParams.containsAll(Arrays.asList(ctor.getParameterTypes()));
-    }
+        List<Constructor<?>> ctors = Arrays.asList(cl.getConstructors());
+        ctors.sort((x, y) -> x.getParameterCount() - y.getParameterCount());
 
-    /**
-     * @return the eventHandlers
-     */
-    public List<Command> getLoadedCommands() {
-        return loadedCommands;
-    }
+        Constructor<? extends T> fitting = null;
 
-    /**
-     * @param eventHandlers the eventHandlers to set
-     */
-    public void setLoadedCommands(List<Command> loadedCommands) {
-        this.loadedCommands = loadedCommands;
+        for (var ctor : ctors) {
+            Class<?>[] params = ctor.getParameterTypes();
+            if (acceptableParams.containsAll(Arrays.asList(ctor.getParameterTypes()))) {
+                try {
+                    return cl.getConstructor(params);
+                } catch (NoSuchMethodException e) {
```

Comment:
Is this github or is that really a syntax error here? Where is the closing parenthesis for that catch?

EDIT: Oh. If you want to swallow it, rename `e` to `ignored` or sth that conveys your intent a bit better.